Source/Request.swift

```diff
@@ -1000,26 +1134,79 @@ public class DownloadRequest: Request {
         return session.downloadTask(with: request)
     }
 
+    /// Creates a `URLSessionTask` from the provided resume data.
+    ///
+    /// - Parameters:
+    ///   - data:    `Data` used to resume the download.
+    ///   - session: `URLSession` used to create the `URLSessionTask`.
+    /// - Returns:   The `URLSessionTask` created.
     public func task(forResumeData data: Data, using session: URLSession) -> URLSessionTask {
         return session.downloadTask(withResumeData: data)
     }
 
+    /// Cancels the instance. Once cancelled, a `DownloadRequest` can no longer be resumed or suspended.
+    ///
+    /// - Note: This method will NOT produce resume data. If you wish to cancel and produce resume data, use
+    ///         `cancel(byProducingResumeData:).
+    ///
+    /// - Returns: The instance.
     @discardableResult
     public override func cancel() -> Self {
+        return cancel(producingResumeData: false)
+    }
+
+    /// Cancels the instance, optionally producing resume data. Once cancelled, a `DownloadRequest` can no longer be
+    /// resumed or suspended.
+    ///
+    /// - Note: If `producingResumeData` is `true`, the `resumeData` property will be populated with any resume data, if
+    ///         available.
+    ///
+    /// - Returns: The instance.
+    public func cancel(producingResumeData shouldProduceResumeData: Bool) -> Self {
```

Comment:
Added a method to cancel without a completion handler but still producing resume data. It's a bit redundant with `cancel()`, but we have to override that. If you can find a better way to call down into one cancellation implementation, that would be great. For now I just pass an empty closure in the `true` case.

Reply:
I can't think of a better approach since you have to `override` the `cancel` API. A better option would be to have a default value of `producingResumeData`, but as you mentioned, you can't override `cancel` using that approach. I think what you have is the only option. Looks good.